server/src/main/java/io/spine/examples/shareaware/server/investment/SharesPurchaseProcess.java

```diff
@@ -0,0 +1,219 @@
+                .vBuild();
+    }
+
+    private void initState(PurchaseShares c) {
+        builder()
+                .setId(c.getPurchaseProcess())
+                .setPurchaser(c.getPurchaser())
+                .setShare(c.getShare())
+                .setQuantity(c.getQuantity());
+    }
+
+    /**
+     * Terminates the process when there are insufficient funds
+     * for shares purchased in the wallet.
+     */
+    @React
+    SharesPurchaseFailed on(InsufficientFunds r) {
+        setArchived(true);
+        return SharesPurchaseFailed
+                .newBuilder()
+                .setPurchaseProcess(r.purchaseProcess())
+                .setPurchaser(state().getPurchaser())
+                .vBuild();
+    }
+
+    /**
+     * Issues the command to obtain shares from the market
+     * after money for it was reserved.
+     */
+    @Command
+    ObtainShares on(MoneyReserved e) {
+        return ObtainShares
+                .newBuilder()
+                .setPurchase(e.purchaseProcess())
+                .setShare(state().getShare())
+                .setQuantity(state().getQuantity())
+                .setMarket(MarketProcess.ID)
+                .vBuild();
+    }
+
+    /**
+     * Issues the command to add shares to the user's investment
+     * after they were bought from the market.
+     */
+    @Command
+    AddShares on(SharesObtained e) {
+        return AddShares
+                .newBuilder()
+                .setInvestment(investmentId())
+                .setProcess(e.getPurchaseProcess())
+                .setQuantity(e.getQuantity())
+                .vBuild();
+    }
+
+    /**
+     * Issues the command to cancel money reservation that was made for shares purchase
+     * after the unexpected error in shares market.
+     */
+    @Command
+    CancelMoneyReservation on(SharesCannotBeObtained r) {
+        return CancelMoneyReservation
+                .newBuilder()
+                .setWallet(walletId())
+                .setOperation(operationId())
+                .vBuild();
+    }
+
+    /**
+     * Terminates the process after the money reservation for shares purchase was cancelled.
+     */
+    @React
+    SharesPurchaseFailed on(MoneyReservationCanceled e) {
+        setArchived(true);
+        return SharesPurchaseFailed
+                .newBuilder()
+                .setPurchaseProcess(state().getId())
+                .setPurchaser(state().getPurchaser())
+                .vBuild();
+    }
+
+    /**
+     * Issues the command to debit the reserved money for shares purchase
+     * after the shares were bought from the market.
+     */
+    @Command
+    DebitReservedMoney on(SharesAdded e) {
+        return DebitReservedMoney
+                .newBuilder()
+                .setWallet(walletId())
+                .setOperation(operationId())
+                .vBuild();
+    }
+
+    /**
+     * Ends the process successfully when reserved money was debited from the wallet.
```

Comment:
Technically, such sentences should look like this:
```
Ends the process successfully when reserved money are debited from the wallet.
```

Please do some research in terms of English grammar. Here is [one thing](https://learnenglish.britishcouncil.org/grammar/english-grammar-reference/verbs-time-clauses-if-clauses#:~:text=In%20time%20clauses%20with%20words,after%20they%20have%20had%20dinner.) to start from.